keylime/agentstates.py

```diff
@@ -6,27 +6,46 @@
 
 import threading
 
-from keylime.ima_ast import START_HASH, FF_HASH
+from keylime.ima_ast import get_FF_HASH, get_START_HASH
+from keylime.common.algorithms import Hash
 from keylime.ima_file_signatures import ImaKeyrings
 
 class TPMState():
     """ TPMState models the state of the TPM's PCRs """
+
     def __init__(self):
         """ constructor """
         self.pcrs = {}
+        self.hash_alg = {}
```

Comment:
Add this comment (or something like it):
# Record the hash_alg that a given PCR is using

I hope they will all be using the same hash algo...

Reply:
Done. Yeah we should enforce that the agent cannot switch algorithm after initial attestation.